### PR TITLE
Added scalatests for tokenService and loginService in example

### LINKED ISSUE #68: Add scalatests for borderpatrol example

### PR DESCRIPTION
**Pull request by maheshkelkar**
Fixes #68

### CODE REVIEW

**Comment by maheshkelkar** [comment]
@trane did you get a chance to review this?

File: example/src/main/scala/com/lookout/borderpatrol/example/reader.scala [hunk 1]
```diff
@@ -60,8 +60,8 @@ object reader {
   val sessionIdReader: RequestReader[SessionId] =
     cookie("border_session").map(_.value).as[SessionId]
 
-  val sessionReader: RequestReader[Session[Request]] =
-    sessionIdReader.embedFlatMap(sessionStore.get[Request]).embedFlatMap {
+  def sessionReader[A](implicit ev: SessionDataEncoder[A]): RequestReader[Session[A]] =
+    sessionIdReader.embedFlatMap(sessId => sessionStore.get[A](sessId)(ev)).embedFlatMap {
```
Comment (trane) [inline example/src/main/scala/com/lookout/borderpatrol/example/reader.scala]: you shouldn't need to explicitly pass the `ev` here

File: example/src/main/scala/com/lookout/borderpatrol/example/service.scala [hunk 2]
```diff
@@ -68,14 +68,18 @@ object service {
     }
 
     def login(req: Request): Future[Response] =
+    {
       (for {
         u <- param("username")(req)
         p <- param("password")(req)
-        s <- sessionReader(req)
-        tr <- tokenService(Request("e" -> u, "p" -> p, "s" -> "login"))
+        s <- sessionReader[String](SessionDataEncoder.encodeString)(req)
```
Comment (trane) [inline example/src/main/scala/com/lookout/borderpatrol/example/service.scala]: this is not what the original code did, the original code read a `Request` to hydrate the original request object (where to redirect them after login is successful).

this should be: `a <- sessionReader(req)` which will be inferred by the compiler as `sessionReader[Request](req)` which will infer the `SessionDataEncoder[Request]` instance to handle the bytes coming from the session store.

File: example/src/main/scala/com/lookout/borderpatrol/example/service.scala [hunk 3]
```diff
@@ -119,11 +125,12 @@ object service {
   case class TokenService(auth: Map[User, Set[String]]) extends Service[Request, Response] {
 
     def apply(req: Request): Future[Response] =
+    {
```
Comment (trane) [inline example/src/main/scala/com/lookout/borderpatrol/example/service.scala]: why add these extra `{}`?

File: example/src/test/scala/com/lookout/borderpatrol/example/ServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package com.lookout.borderpatrol.example
+
+import java.util
+//import java.util.NoSuchElementException
```
Comment (trane) [inline example/src/test/scala/com/lookout/borderpatrol/example/ServiceSpec.scala]: any reason to leave commented-out imports?

File: example/src/test/scala/com/lookout/borderpatrol/example/ServiceSpec.scala [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+//import com.lookout.borderpatrol.sessionx.SessionId.SessionIdInjections
+import com.lookout.borderpatrol.sessionx._
+import com.twitter.finagle.httpx._
+import com.twitter.util._
+import org.scalatest.{OptionValues, TryValues, Matchers, FlatSpec}
+//import io.finch.response.{Forbidden, ResponseBuilder, Ok}
+
+import com.lookout.borderpatrol.example.service._
+
+class ServiceSpec extends FlatSpec with Matchers with TryValues with OptionValues {
```
Comment (trane) [inline example/src/test/scala/com/lookout/borderpatrol/example/ServiceSpec.scala]: this should use the `BorderPatrolSuite` for tests

File: example/src/test/scala/com/lookout/borderpatrol/example/ServiceSpec.scala [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    val cooki = identity[Cookie](id)
+    val session_data = "mvk"
+    sessionStore.update[String](Session(id, session_data))
+
+    // Create request
+    val request = Request(Method.Post, uri)
+    request.addCookie(cooki)
+
+    val response = loginService(request)
+    Await.result(response).status should be (Status.TemporaryRedirect)
```
Comment (trane) [inline example/src/test/scala/com/lookout/borderpatrol/example/ServiceSpec.scala]: we want to ensure that the location we are sending the user is the one they had requested previously.

**Comment by maheshkelkar** [comment]
No plans to merge this. As this was just a work in progress and already replaced by new code.

**Pull request closed without merging**